frontend/test/metabase/scenarios/models/reproductions/20963-can-not-convert-question-with-snippets-to-model.cy.spec.js

```diff
@@ -0,0 +1,58 @@
+import { restore, modal } from "__support__/e2e/cypress";
+import {
+  openDetailsSidebar,
+  getDetailsSidebarActions,
+} from "../helpers/e2e-models-helpers";
+
+const snippetName = `string 'test'`;
+const questionName = "Converting questions with snippets to models";
+
+describe("issue 20963", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("should allow converting questions with static snippets to models (metabase#20963)", () => {
+    cy.visit("/");
+
+    cy.findByText("New").click();
+    cy.findByText("SQL query").click();
+
+    // Creat a snippet
+    cy.icon("snippet").click();
+    cy.findByText("Create a snippet").click();
+
+    modal().within(() => {
+      cy.findByLabelText("Enter some SQL here so you can reuse it later").type(
+        `'test'`,
+      );
+      cy.findByLabelText("Give your snippet a name").type(snippetName);
+      cy.findByText("Save").click();
+    });
+
+    cy.get(".ace_text-input").type("{home}select ", {
+      force: true,
+    });
+
+    cy.findByText("Save").click();
+    modal().within(() => {
+      // I don't know why the input lost focus when the delay is defaulted to 10ms
+      cy.findByLabelText("Name").type(questionName);
```

Comment:
I found there's some weird behavior that makes the input lose focus, especially if we ran the query before saving as a question.

Reply:
Sounds like #18150

Reply:
That's exactly what happened.